jt_notes.md

```diff
@@ -0,0 +1,127 @@
+
+prospective_optical_gating.py
+
+[No comments beyond the inline ones in the source code]
+
+emulate.py
+
+I haven't really read through this in detail yet, but I do remember that I didn't find it particularly helpful as an example to follow when I was trying to do my own offline analysis using this codebase. I need to go back and remind myself what the difficulties were that I encountered...
+
+stage_control_functions.py
+
+Looks pretty well commented and structured, but I would implement this as a class, so that other (sub-)classes could be substituted to control other stage types, etc. I will give this some more thought. **For me**: how is this used? Which object is the most sensible one to be in charge of all this?
+
+app
+
+For my benefit: base_camera.py runs a background thread which posts events whenever a frame is available. frames() uses "yield" to pass frame buffers to an iterator.
+
+app.gen() is a "video streaming generator function" which in turn "yield"s frames (in what looks like a MIME-like format) to the caller. I am not clear how video_feed behaves, though.
+
+Basic questions **for chas**:
+
+- What is the overall function of app.py? Do I understand correctly that these decorated functions are called in response to messages from the client webpage? In response to those messages, they appear to both perform internal state-affecting actions (i.e. run the synchronization algorithms, in some cases), *and also* then return the actual HTTP source(*) that should now be displayed in the webpage. Am I understanding that correctly? What is the deal with the "session" object? Could really do with a comment at the top of the module to explain what's going on with all of this.                (*) Oh, or sometimes raw video images apparently, in video_feed()...
+
+- **IMPORTANT:** When app.run() refers to "live", is that what actually gets run for live sync with a microscope, fish and everything? Because, for instance, it calls through to **emulate**_get_period...!? I don't know if this is because, as you said, the app code needs updating, but I can't see how it is meaningfully interacting with cli in the case of **live** frames, and I can't quite picture how that even *should* work. It seems like the gen() function should be being used somehow, but that's not being used except in video_feed().
+
+- base_camera.py needs a similar comment at the top of the module explaining what's going on - and in particular what the relationship is between a "client" and a "thread", explanation of the role of "ident" (which is a unique identifier associated with each thread), etc. Are these concurrent threads or [I suspect] sequential threads? Really needs an explanation of the thread architecture here, and the role of "yield" in terms how of the threads interact.
```

Comment:
This was not altered from the original repo. But, may also not be the way we want to go (c.f. the imutils way)